diff --git a/dist/index.js b/dist/index.js
index 724cd6fcdf2dce8622bcef6a4880e92fe05cf44c..94978da5cd64760ef09b5f5020e31d671898db8c 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -63,11 +63,17 @@ var SlotClone = React.forwardRef((props, forwardedRef) => {
   const { children, ...slotProps } = props;
   if (React.isValidElement(children)) {
     const childrenRef = getElementRef(children);
-    return React.cloneElement(children, {
+    // TODO https://github.com/radix-ui/primitives/pull/3229
+    const props = {
       ...mergeProps(slotProps, children.props),
       // @ts-ignore
       ref: forwardedRef ? (0, import_react_compose_refs.composeRefs)(forwardedRef, childrenRef) : childrenRef
-    });
+    };
+    // do not pass ref to React.Fragment for React 19 compatibility
+    if (children.type !== React.Fragment) {
+      props.ref = forwardedRef ? (0, import_react_compose_refs.composeRefs)(forwardedRef, childrenRef) : childrenRef;
+    }
+    return React.cloneElement(children, props);
   }
   return React.Children.count(children) > 1 ? React.Children.only(null) : null;
 });
diff --git a/dist/index.mjs b/dist/index.mjs
index 73e28afb741e05e1d1bb4206fa917918bb9457ec..b455ee34d5f8eadf0f454460574d6995ea9a74be 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -25,11 +25,17 @@ var SlotClone = React.forwardRef((props, forwardedRef) => {
   const { children, ...slotProps } = props;
   if (React.isValidElement(children)) {
     const childrenRef = getElementRef(children);
-    return React.cloneElement(children, {
+    // TODO https://github.com/radix-ui/primitives/pull/3229
+    const props = {
       ...mergeProps(slotProps, children.props),
       // @ts-ignore
       ref: forwardedRef ? composeRefs(forwardedRef, childrenRef) : childrenRef
-    });
+    };
+    // do not pass ref to React.Fragment for React 19 compatibility
+    if (children.type !== React.Fragment) {
+      props.ref = forwardedRef ? composeRefs(forwardedRef, childrenRef) : childrenRef;
+    }
+    return React.cloneElement(children, props);
   }
   return React.Children.count(children) > 1 ? React.Children.only(null) : null;
 });
